linera-execution/src/system.rs

```diff
@@ -608,45 +578,15 @@ where
                 ))?;
             }
             CreateApplication {
-                bytecode_id,
-                parameters,
-                instantiation_argument,
-                required_application_ids,
-            } => {
-                let id = UserApplicationId {
-                    bytecode_id,
-                    creation: context.next_message_id(txn_tracker.next_message_index()),
-                };
-                self.registry
-                    .register_new_application(
-                        id,
-                        parameters.clone(),
-                        required_application_ids.clone(),
-                    )
-                    .await?;
-                // Send a message to ourself to increment the message ID.
-                let message = RawOutgoingMessage {
-                    destination: Destination::Recipient(context.chain_id),
-                    authenticated: false,
-                    grant: Amount::ZERO,
-                    kind: MessageKind::Protected,
-                    message: SystemMessage::ApplicationCreated,
-                };
-                outcome.messages.push(message);
-                new_application = Some((id, instantiation_argument.clone()));
-            }
-            RequestApplication {
-                chain_id,
                 application_id,
+                instantiation_argument,
             } => {
-                let message = RawOutgoingMessage {
-                    destination: Destination::Recipient(chain_id),
-                    authenticated: false,
-                    grant: Amount::ZERO,
-                    kind: MessageKind::Simple,
-                    message: SystemMessage::RequestApplication(application_id),
-                };
-                outcome.messages.push(message);
+                txn_tracker.replay_oracle_response(OracleResponse::Blob(
+                    BlobId::new_application_description_from_hash(
+                        application_id.application_description_hash,
```

Comment:
What if the application description contains a wrong chain ID, or height, or effect counter value?

Wouldn't it be better to really _create_ the blob here, and have the operation only contain the bytecode, parameters and required application IDs?

Reply:
OK; yes, if we have to create the blob on the client side, we must verify it here.